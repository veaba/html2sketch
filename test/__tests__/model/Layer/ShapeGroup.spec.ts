import { ShapeGroup } from 'html2sketch';
import { describe, test, expect, it, beforeAll  } from 'vitest'

describe('ShapeGroup', () => {
  describe('toSketchJSON', () => {
    it('生成正常', () => {
      const shapeGroup = new ShapeGroup();
      expect(shapeGroup.toSketchJSON()).toMatchSnapshot();
    });
  });
  it('生成正常', () => {
    const shapeGroup = new ShapeGroup();
    expect(shapeGroup).toMatchSnapshot();
  });

  it('设置剪贴蒙版', () => {
    const shapeGroup = new ShapeGroup();

    expect(shapeGroup.toSketchJSON().hasClippingMask).toBe(false);

    shapeGroup.hasClippingMask = true;
    expect(shapeGroup.toSketchJSON().hasClippingMask).toBe(true);
  });
});
