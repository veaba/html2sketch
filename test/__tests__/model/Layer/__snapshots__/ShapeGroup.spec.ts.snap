// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ShapeGroup > toSketchJSON > 生成正常 1`] = `
{
  "_class": "shapeGroup",
  "booleanOperation": -1,
  "clippingMaskMode": 0,
  "do_objectID": "UUID",
  "exportOptions": {
    "_class": "exportOptions",
    "exportFormats": [],
    "includedLayerIds": [],
    "layerOptions": 0,
    "shouldTrim": false,
  },
  "frame": {
    "_class": "rect",
    "constrainProportions": false,
    "height": 0,
    "width": 0,
    "x": 0,
    "y": 0,
  },
  "hasClickThrough": false,
  "hasClippingMask": false,
  "isFixedToViewport": false,
  "isFlippedHorizontal": false,
  "isFlippedVertical": false,
  "isLocked": false,
  "isTemplate": false,
  "isVisible": true,
  "layerListExpandedType": 0,
  "layers": [],
  "name": "shapeGroup",
  "nameIsFixed": false,
  "resizingConstraint": 27,
  "resizingType": 0,
  "rotation": 0,
  "shouldBreakMaskChain": false,
  "style": {
    "_class": "style",
    "borderOptions": {
      "_class": "borderOptions",
      "dashPattern": [],
      "isEnabled": true,
      "lineCapStyle": 0,
      "lineJoinStyle": 0,
    },
    "borders": [],
    "colorControls": {
      "_class": "colorControls",
      "brightness": 0,
      "contrast": 1,
      "hue": 0,
      "isEnabled": false,
      "saturation": 1,
    },
    "contextSettings": {
      "_class": "graphicsContextSettings",
      "blendMode": 0,
      "opacity": 1,
    },
    "do_objectID": "UUID",
    "endMarkerType": 0,
    "fills": [],
    "innerShadows": [],
    "miterLimit": 10,
    "shadows": [],
    "startMarkerType": 0,
    "windingRule": 1,
  },
  "windingRule": 1,
}
`;

exports[`ShapeGroup > 生成正常 1`] = `
{
  "id": "UUID",
  "layers": [],
  "locked": false,
  "name": "",
  "resizingConstraint": 27,
}
`;
